src/cljs/orcpub/dnd/e5/views.cljs

```diff
@@ -2904,6 +2904,35 @@
                            weight)]]]]))
               equipment-cfgs))]]]]))))
 
+(defn treasure-section []
+  (let [expanded-details (r/atom {})]
+    (fn [id]
```

Comment:
If all you're doing with a Form-2 component is creating ratoms before returning the rendering fn, you should prefer reagent's [`with-let`](https://cljdoc.org/d/reagent/reagent/0.9.0-SNAPSHOT/doc/tutorials/creating-reagent-components#appendix-b---with-let-macro) macro instead.

It's less error-prone for the reason seen here: there's a mismatch between the params passed into the component fn and the returned render fn 😄  (`id` is in the render fn, but not the top-level `treasure-section`)

Reply:
Heh, if you're using `with-let`, you won't return a render function. It goes from a reagent Form-2 to a Form-1. And you'll need to add the missing `id` param to the main component fn.

Like:

```clojure
(defn treasure-section [id]
  (r/with-let [expanded-details (r/atom {})]
    (let [mobile? @(subscribe [:mobile?])
          treasure-cfgs @(subscribe [::char/treasure id])]
      ...
```